From b9f7a8dc0769bddfb2dff47a0903b51471ebf4ef Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Fri, 24 Sep 2010 18:02:42 +0200
Subject: [PATCH 2/2] Use CFLAGS_MODULE together with MODFLAGS in make.sh

CFLAGS_MODULE was introduced in 2.6.36 kernels while MODFLAGS was
dropped (commit 6588169d516560f68672e2928680b71c647b7806) therefore
by trying to use both modules we preserve compatibility with older
kernels and fix build issues with new kernels.

Signed-off-by: Alberto Milone <alberto.milone@canonical.com>

[Adapted for Mandriva build / tmb]
Signed-off-by: Thomas Backlund <tmb@mandriva.org>
[anssi@mandriva.org: adapted for upstream changes]
Signed-off-by: Anssi Hannula <anssi@mandriva.org>

---

--- fglrx-8.771.orig/common/lib/modules/fglrx/build_mod/make.sh	2010-10-02 20:10:04.000000000 +0300
+++ fglrx-8.771/common/lib/modules/fglrx/build_mod/make.sh	2010-10-02 20:20:07.462928823 +0300
@@ -473,9 +473,12 @@ make clean
 
 echo 'This is a dummy file created to suppress this warning: could not find /lib/modules/fglrx/build_mod/2.6.x/.libfglrx_ip.a.GCC4.cmd for /lib/modules/fglrx/build_mod/2.6.x/libfglrx_ip.a.GCC4' > .lib${MODULE}_ip.a.GCC${GCC_MAJOR}.cmd
 
+    MODFLAGS="-DMODULE -DATI -DFGL -DPAGE_ATTR_FIX=$PAGE_ATTR_FIX -DCOMPAT_ALLOC_USER_SPACE=$COMPAT_ALLOC_USER_SPACE $def_smp $def_modversions"
+
 make CC=${CC} \
     LIBIP_PREFIX=$(echo "$LIBIP_PREFIX" | sed -e 's|^\([^/]\)|../\1|') \
-    MODFLAGS="-DMODULE -DATI -DFGL -DPAGE_ATTR_FIX=$PAGE_ATTR_FIX -DCOMPAT_ALLOC_USER_SPACE=$COMPAT_ALLOC_USER_SPACE $def_smp $def_modversions" \
+    MODFLAGS="$MODFLAGS" \
+    CFLAGS_MODULE="$MODFLAGS" \
     KVER=${uname_r} \
     KDIR=${kernel_dir} \
     PAGE_ATTR_FIX=$PAGE_ATTR_FIX \
